lib/elixir_analyzer/exercise_test/assert_call/compiler.ex

```diff
@@ -81,15 +82,10 @@ defmodule ElixirAnalyzer.ExerciseTest.AssertCall.Compiler do
       |> track_aliases(node)
       |> track_imports(node)
 
-    function_def? = function_def?(node)
-    name = extract_function_name(node)
-
-    case {function_def?, name} do
-      {true, name} ->
-        {node, %{acc | in_function_def: name}}
-
-      _ ->
-        {node, acc}
+    cond do
+      module_def?(node) -> {node, %{acc | in_module: extract_module_name(node)}}
+      function_def?(node) -> {node, %{acc | in_function_def: extract_function_name(node)}}
+      true -> {node, acc}
```

Comment:
In Haskell, the keyword `otherwise` is an alias for `True`, it makes this kind of check a lot more readable. I always find awkward writing this
```elixir
cond do
  this?(x) -> also_that?(x)
  that?(x) -> also_this?(x)
  true -> true
end
```
I think this is much smoother:
```elixir
cond do
  this?(x) -> also_that?(x)
  that?(x) -> also_this?(x)
  otherwise -> true
end
```
Do you think that's a valid feature request? :)

Reply:
Haha 😀. I'm not sure about Haskell, but in Elixir `true` is just syntactic sugar for the atom `:true`. You can already write:

```elixir
cond do
  this?(x) -> also_that?(x)
  that?(x) -> also_this?(x)
  :otherwise -> true
end
```

Because all atoms are truthy and `cond` doesn't require a strictly boolean expression.

There are only three atoms that have this extra treatment that you can write them without `:` - `true`, `false`, and `nil`. I'm not sure if it's such a pressing problem that they would want to add `otherwise` to that list 😅. But I do see the problems. I hate conds that look like this:

```elixir
cond do
  something == false -> :ok
  true -> :error
end
```

The pairing of `false` and `true` here makes my brain hurt.

Reply:
I would not call that a pressing problem, no, but maybe a nice little quality of life update, like Elixir does :)

`:otherwise -> true` is... almost there, but not quite. It might appease me, but definitely not most people who would need to use my code after me!

I share your brain pain!